.github/workflows/main.yml

```diff
@@ -0,0 +1,66 @@
+name: comet-common CI
+
+on:
+  # Trigger the workflow on all push, pull request for the master branch and release events
+  push:
+  pull_request:
+    branches: [ master ]
+    types: [closed]
+  release:
+    type: [published]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.6, 3.7, 3.8, 3.9]
+        include: # define single matrix case that performs the upload
+          - os: ubuntu-latess
+            python: 3.6
+            upload: true
+    steps:
+      # Checks-out the repository under $GITHUB_WORKSPACE
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Cache pip check
```

Comment:
Just curious, what does this step do?

Reply:
This action sets up a Python environment for use by the actions, more details can be found in [here](https://github.com/actions/setup-python).
In our case is responsible for setting up a workflow for each python version what we are supported.